### PR TITLE
T-56 TimeProvider

### PR DESCRIPTION
**Pull request by DashaDrahun**
created timeprovider, updated tests

### CODE REVIEW

File: PortfolioKeeper/App_Start/AutofacConfig.cs [hunk 1]
```diff
@@ -87,6 +87,10 @@ private static void RegisterBusinessComponents(ContainerBuilder builder)
                 .As<IIngotPriceProvider>()
                 .InstancePerLifetimeScope();
 
+            builder.RegisterType<TimeProvider>()
+                .As<ITimeProvider>()
+                .InstancePerLifetimeScope();
```
Comment (AntonBessarabov) [inline PortfolioKeeper/App_Start/AutofacConfig.cs]: .SingleInstance()

File: PortfolioKeeper/BusinessLogic/Mappers/AssetMapper.cs [hunk 2]
```diff
@@ -10,11 +11,14 @@ namespace PortfolioKeeper.BusinessLogic.Mappers
     public class AssetMapper : IAssetMapper
     {
         private readonly IAssetPropertyMapper propertyTypeValueMapper;
+        private readonly ITimeProvider timeProvider;
 
-        public AssetMapper(IAssetPropertyMapper propertyTypeValueMapper)
+        public AssetMapper(IAssetPropertyMapper propertyTypeValueMapper, ITimeProvider timeProvider)
```
Comment (stas-litvinenka) [inline PortfolioKeeper/BusinessLogic/Mappers/AssetMapper.cs]: to new line

File: PortfolioKeeper/BusinessLogic/Mappers/AssetMapper.cs [hunk 3]
```diff
@@ -30,8 +34,8 @@ public Asset Map(string name, int portfolioId, byte assetTypeId, IEnumerable<Ass
             {
                 Name = name,
                 PortfolioId = portfolioId,
-                CreatedOn = date,
-                UpdatedOn = date,
+                CreatedOn = timeProvider.GetUtcTime(),
```
Comment (stas-litvinenka) [inline PortfolioKeeper/BusinessLogic/Mappers/AssetMapper.cs]: better to create variable, then assign it value to both - CreatedOn ,UpdatedOn

File: PortfolioKeeper/BusinessLogic/Services/PortfolioService.cs [hunk 4]
```diff
@@ -41,7 +46,7 @@ public void Create(PortfolioDto portfolioDto)
             try
             {
                 var newPlot = mapper.Map<PortfolioDto, Portfolio>(portfolioDto);
-                newPlot.CreatedOn = newPlot.UpdatedOn = DateTime.UtcNow;
+                newPlot.CreatedOn = newPlot.UpdatedOn = timeProvider.GetUtcTime();
```
Comment (stas-litvinenka) [inline PortfolioKeeper/BusinessLogic/Services/PortfolioService.cs]: create variable, then assign to both CreatedOn ,UpdatedOn

File: PortfolioKeeper/BusinessLogic/Services/PortfolioService.cs [hunk 5]
```diff
@@ -46,7 +46,9 @@ public void Create(PortfolioDto portfolioDto)
             try
             {
                 var newPlot = mapper.Map<PortfolioDto, Portfolio>(portfolioDto);
-                newPlot.CreatedOn = newPlot.UpdatedOn = timeProvider.GetUtcTime();
+                var date = timeProvider.GetUtcTime();
+
+                newPlot.CreatedOn = newPlot.UpdatedOn = date;
```
Comment (stas-litvinenka) [inline PortfolioKeeper/BusinessLogic/Services/PortfolioService.cs]: slit. newPlot.CreatedOn=date; newPlot.UpdatedOn =date;

File: PortfolioKeeper.Tests/BusinessLogic/Mappers/AssetMapperTests.cs [hunk 6]
```diff
@@ -30,6 +30,8 @@ public void SetUp()
                 Name = "FirstAsset",
                 PortfolioId = 1
             };
+            expectedDateTime = new DateTime(2018, 7, 16);
+            timeProvider.Setup(f => f.GetUtcTime()).Returns(expectedDateTime);
```
Comment (stas-litvinenka) [inline PortfolioKeeper.Tests/BusinessLogic/Mappers/AssetMapperTests.cs]: Optional: if you need this mock only in one place you can mock it there. not in SetUp

**Pull request merged**